recipes/cad1/task1/baseline/enhance.py

```diff
@@ -0,0 +1,304 @@
+        mix (torch.Tensor): mixture to separate, shape (batch, channels, time)
+        sample_rate (int): sampling rate of the mixture
+        segment (float): segment length in seconds
+        overlap (float): overlap between segments, between 0 and 1
+        device (torch.device, str, or None): if provided, device on which to
+            execute the computation, otherwise `mix.device` is assumed.
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+
+    Returns:
+        torch.Tensor: estimated sources
+
+    Based on https://pytorch.org/audio/main/tutorials/hybrid_demucs_tutorial.html
+    """
+    device = mix.device if device is None else torch.device(device)
+    mix = torch.as_tensor(mix, device=device)
+
+    if mix.ndim == 1:
+        # one track and mono audio
+        mix = mix.unsqueeze(0)
+    elif mix.ndim == 2:
+        # one track and stereo audio
+        mix = mix.unsqueeze(0)
+
+    batch, channels, length = mix.shape
+
+    chunk_len = int(sample_rate * segment * (1 + overlap))
+    start = 0
+    end = chunk_len
+    overlap_frames = overlap * sample_rate
+    fade = Fade(fade_in_len=0, fade_out_len=int(overlap_frames), fade_shape="linear")
+
+    final = torch.zeros(batch, len(model.sources), channels, length, device=device)
+
+    while start < length - overlap_frames:
+        chunk = mix[:, :, start:end]
+        with torch.no_grad():
+            out = model.forward(chunk)
+        out = fade(out)
+        final[:, :, :, start:end] += out
+        if start == 0:
+            fade.fade_in_len = int(overlap_frames)
+            start += int(chunk_len - overlap_frames)
+        else:
+            start += chunk_len
+        end += chunk_len
+        if end >= length:
+            fade.fade_out_len = 0
+
+    return final.cpu().detach().numpy()
+
+
+def get_device(device: str) -> tuple:
+    """Get the Torch device.
+
+    Args:
+        device (str): device type, e.g. "cpu", "gpu0", "gpu1", etc.
+
+    Returns:
+        torch.device: torch.device() appropiate to the hardware available.
+        str: device type selected, e.g. "cpu", "cuda".
+    """
+    if device is None:
+        if torch.cuda.is_available():
+            return torch.device("cuda"), "cuda"
+        return torch.device("cpu"), "cpu"
+
+    elif device.startswith("gpu"):
+        device_index = int(device.replace("gpu", ""))
+        if device_index > torch.cuda.device_count():
+            raise ValueError(f"GPU device index {device_index} is not available.")
+        return torch.device(f"cuda:{device_index}"), "cuda"
+
+    elif device == "cpu":
+        return torch.device("cpu"), "cpu"
+
+    else:
+        raise ValueError(f"Unsupported device type: {device}")
+
+
+def map_to_dict(sources: np.ndarray, sources_list: List[str]) -> Dict:
+    """Map sources to a dictionary."""
```

Comment:
Docstring suggestion...

```suggestion
    """Map sources to a dictionary separating audio into left and right channels.
    
    Args:
       sources (np.ndarray): Signal to be mapped to dictionary.
       sources_list (list): List of strings used to index dictionary.
       
    Returns:
        Dictionary: A dictionary of separated source audio split into channels.
    """
```